ui/server/src/main/scala/pl/touk/nussknacker/ui/uiresolving/UIProcessResolving.scala

```diff
@@ -0,0 +1,42 @@
+package pl.touk.nussknacker.ui.uiresolving
+
+import pl.touk.nussknacker.engine.ProcessingTypeData.ProcessingType
+import pl.touk.nussknacker.engine.api.deployment.GraphProcess
+import pl.touk.nussknacker.engine.api.expression.ExpressionTypingInfo
+import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
+import pl.touk.nussknacker.engine.dict.ProcessDictSubstitutor
+import pl.touk.nussknacker.engine.marshall.ProcessMarshaller
+import pl.touk.nussknacker.engine.spel.SpelExpressionParser
+import pl.touk.nussknacker.restmodel.displayedgraph.{DisplayableProcess, ValidatedDisplayableProcess}
+import pl.touk.nussknacker.restmodel.validation.ValidationResults.ValidationResult
+import pl.touk.nussknacker.ui.process.marshall.ProcessConverter
+import pl.touk.nussknacker.ui.validation.ProcessValidation
+
+/**
+  * This class handles resolving of expression (e.g. dict labels resolved to keys) which should be done before
+  * validation of process created in UI and before process save.
+  * Also it handles "reverse" resolving process done before returning process to UI
+  */
+class UIProcessResolving(validation: ProcessValidation, dictSubstitutor: ProcessDictSubstitutor) {
```

Comment:
let's add some TODO to refactor that :)

Reply:
But where? I think that real TODO should be in each `Resource`, and in `CanonicalProcess` (it is there). Here we have only result of missing refactoring in other places ;-)